packages/contentful-app-extensions/date-field-as-first-published-at/src/locations/Field.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+import { Text } from '@contentful/f36-components';
+import { EntrySys, FieldExtensionSDK } from '@contentful/app-sdk';
+import { useSDK } from '@contentful/react-apps-toolkit';
+
+const Field = () => {
+  const sdk = useSDK<FieldExtensionSDK>();
+
+  const [field, setField] = useState(sdk.field.getValue());
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  let unsubscribe = () => {};
+
+  unsubscribe = sdk.entry.onSysChanged(async (sys: EntrySys) => {
+    if (!field && sys.firstPublishedAt) {
+      setField(sys.firstPublishedAt);
+      await sdk.field.setValue(sys.firstPublishedAt);
+      await sdk.entry.publish();
```

Comment:
Why are we publishing the entry? I thought the user had to press the publish button on the form

Reply:
The flow would be something like
- The user publishes the entry
- The field is set with the value `firstPublishedAt`
- The entry status becomes changed

if I don't publish in this line 18 the status remains as changed. With this `publish` the status becomes `published` without the user having to publish two times in a row :)